File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let n = progresses.length;
    
    const doneDay = Array(n).fill(0);
    
    for(let i = 0; i<n; i++){
        doneDay[i] = Math.ceil((100 - progresses[i]) / speeds[i])
    }
    
    let currentDay = doneDay[0];
    let count = 1;
    
    for(let i = 1; i<n; i++){
        if(doneDay[i] <= currentDay){
            count ++;
        } else{
            answer.push(count);
            count = 1;
            currentDay = doneDay[i];
        }
    }
    
    answer.push(count);
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds):
    answer = []
    done = [math.ceil((100 - p) / s) for p, s in zip(progresses, speeds)]
    
    currentDay = done[0]
    count = 1
    
    for d in done[1:]:
        if d <= currentDay:
            count+=1
        else:
            answer.append(count)
            count = 1
            currentDay = d
        
    answer.append(count)
    return answer